bundles/framework/admin-users/view/RolesTab.jsx

```diff
@@ -57,12 +63,43 @@ export const RolesTab = ({ state, controller }) => {
             </Form>
             {roles.map(role => (
                 <RoleBlock key={role.id}>
-                    <span>{role.name}</span>
-                    <DeleteButton
-                        type='button'
-                        title={<Message messageKey='flyout.adminroles.confirm_delete' messageArgs={{ role: role.name }} />}
-                        onConfirm={() => controller.deleteRole(role.id)}
-                    />
+                    {state.editingRole && state.editingRole.id === role.id ? (
+                        <StyledInput
+                            value={state.editingRole.name}
+                            onChange={(e) => controller.updateEditingRole(e.target.value)}
+                            status={state.editingRoleError ? 'error' : null}
+                        />
+                    ) : (
+                        <span>{role.name}</span>
+                    )}
+                    {state.systemRoles.findIndex(r => r === role.name) < 0 && (
```

Comment:
Can we use includes() here? 
```suggestion
                    {!state.systemRoles.includes(r => r === role.name) && (
```
Or maybe even inject a boolean property to roles on the list so we could just check some role.editableRole here?